docs/analysis/perfetto-sql-backcompat.md

```diff
@@ -8,6 +8,92 @@ documents:
  - **Context**: why we are making the change i.e. why does it have to backwards incompatible?
  - **Migrations**: suggested changes you can make to your PerfettoSQL to not be broken by the changes
 
+## Removal of `stack_id` and `parent_stack_id` columns from slice table
+
+**Date/Version**
+
+2025-11-21/v54.0
+
+**Symptoms**
+
+- An error message of the form `no such column: stack_id` or `no such column: parent_stack_id`
+- The `stack_id` and `parent_stack_id` columns disappearing from output of queries with `SELECT *` on the slice table
+- Error `no such function: ancestor_slice_by_stack` or `no such function: descendant_slice_by_stack` (these now require `INCLUDE PERFETTO MODULE slices.stack`)
+- Queries using hardcoded stack_id values (e.g., `WHERE stack_id = 123456`) return no results or wrong results
+- JOINs between old and new traces on stack_id fail to match expected slices
+
+**Context**
+
+The `stack_id` and `parent_stack_id` columns were legacy features that computed hash values based on slice names. These columns were rarely useful in practice and were documented as "kept around only for legacy reasons". The implementation had several issues:
+
+- They consumed non-trivial memory on large traces to store pre-computed hash values
```

Comment:
I think this whole list is not accurate. There are two reasons: 1) they used a lot of memory for a feature which was rarely useful 2) even when they were used, folks should have been using parent_id because they misunderstood what these columns represented.